pkg/v5/resource/chartconfig/desired.go

```diff
@@ -175,6 +185,38 @@ func newNodeExporterChartConfig(clusterConfig cluster.Config, projectName string
 	}
 }
 
+func (r *Resource) getConfigMapSpec(ctx context.Context, guestConfig cluster.Config, configMapName, namespace string) (*v1alpha1.ChartConfigSpecConfigMap, error) {
+	guestK8sClient, err := r.getGuestK8sClient(ctx, guestConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	configMap, err := guestK8sClient.CoreV1().ConfigMaps(namespace).Get(configMapName, apismetav1.GetOptions{})
+	if apierrors.IsNotFound(err) || guest.IsAPINotAvailable(err) {
+		// Cannot get configmap resource version so leave it unset. We will
+		// check again after the next resync period.
+		configMapSpec := &v1alpha1.ChartConfigSpecConfigMap{
+			Name:      configMapName,
+			Namespace: namespace,
+		}
+
+		return configMapSpec, nil
+	} else if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Set the configmap resource version. When this changes it will generate
+	// an update event for chart-operator. chart-operator will recalculate the
+	// desired state including any updated config map values.
+	configMapSpec := &v1alpha1.ChartConfigSpecConfigMap{
+		Name:            configMapName,
+		Namespace:       namespace,
+		ResourceVersion: configMap.ObjectMeta.ResourceVersion,
```

Comment:
What happens when the resource version did not change?  Do we still apply something that is equal? Should we rather consider not doing anything at all then?

Reply:
We compare the spec section and the labels. So if the resource version doesn't change there won't be an update event.

https://github.com/giantswarm/cluster-operator/blob/82ce551f9a93828b8446b497d61cf4ab3af110f4/pkg/v5/resource/chartconfig/resource.go#L159